packages/terra-table/src/Table.jsx

```diff
@@ -1,68 +1,401 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import classNames from 'classnames/bind';
-import TableHeader from './TableHeader';
-import TableHeaderCell from './TableHeaderCell';
-import TableRows from './TableRows';
-import TableRow from './TableRow';
-import TableCell from './TableCell';
-import SelectableTableRows from './SelectableTableRows';
-import TableSingleSelectableRows from './SingleSelectableRows';
-import TableMultiSelectableRows from './MultiSelectableRows';
-import TableSubheader from './TableSubheader';
+import ContentContainer from 'terra-content-container';
+import VisuallyHiddenText from 'terra-visually-hidden-text';
 import styles from './Table.module.scss';
+import sectionShape from './proptypes/sectionShape';
+import headerShape from './proptypes/headerShape';
+import widthShape from './proptypes/widthShape';
+
+import Row from './subcomponents/_Row';
+import Cell from './subcomponents/_Cell';
+import Section from './subcomponents/_Section';
+import HeaderRow from './subcomponents/_HeaderRow';
+import HeaderCell from './subcomponents/_HeaderCell';
+import ChevronCell from './subcomponents/_ChevronCell';
+import CheckMarkCell from './subcomponents/_CheckMarkCell';
+import HeaderChevronCell from './subcomponents/_HeaderChevronCell';
+import HeaderCheckMarkCell from './subcomponents/_HeaderCheckMarkCell';
 
 const cx = classNames.bind(styles);
 
 const propTypes = {
   /**
-   * The children passed to the component
+   * The check mark styling to apply.
+   */
+  checkStyle: PropTypes.oneOf([
+    'none',
+    'readOnly', // TODO: address renaming this, UX ask
+    'toggle',
+  ]),
+  /**
+   * The divider styling to apply to the child rows.
+   */
+  dividerStyle: PropTypes.oneOf(['none', 'vertical', 'horizontal', 'both']),
+  /**
+   * The width value structures associated to each column.
+   */
+  columnWidths: PropTypes.arrayOf(widthShape),
+  /**
+   * Whether or not the rows should have chevrons applied.
+   */
+  hasChevrons: PropTypes.bool,
+  /**
+   * The data to build header cells and columns.
+   */
+  headerData: headerShape,
+  /**
+   * Element to append to the top of the table. i.e. toolbars etc.
+   */
+  headerNode: PropTypes.node,
```

Comment:
I'm not seeing a doc example using `headerNode` or `footerNode`. How are these handled in a restricted height? These aren't sticky are they? Using  `toolbars` as a suggested use case sounds like it would be.

If there is a restricted heigh on the table parent, does are the header/footer squished together for less content room or will it just scroll?

Reply:
The example is under:
Custom Header Footer Node Table

The header/footer behaves the same way as table, block by default, and fixed elements with scrolling content when set to fill. So in a fill=true scenario the header and footer nodes would be sticky.